worker/src/lib/managers/RequestResponseManager.ts

```diff
@@ -0,0 +1,116 @@
+import { SupabaseClient } from "@supabase/supabase-js";
+import { S3Client } from "../clients/S3Client";
+import { err, Result } from "../util/results";
+import { Database } from "../../../supabase/database.types";
+
+export type RequestResponseContent = {
+  requestId: string;
+  organizationId: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  requestBody: any;
+  responseBody: string;
+  requestAssets: Record<string, string>;
+};
+
+export class RequestResponseManager {
+  constructor(
+    private s3Client: S3Client,
+    private supabase: SupabaseClient<Database>
+  ) {}
+
+  async storeRequestResponseData({
+    organizationId,
+    requestId,
+    requestBody,
+    responseBody,
+  }: RequestResponseContent): Promise<Result<string, string>> {
+    const url = this.s3Client.getRequestResponseUrl(requestId, organizationId);
+
+    return await this.s3Client.store(
+      url,
+      JSON.stringify({ requestBody, responseBody })
+    );
+  }
+  async storeRequestResponseImage({
+    organizationId,
+    requestId,
+    requestBody,
+    responseBody,
+    requestAssets,
+  }: RequestResponseContent): Promise<Result<string, string>> {
+    const uploadPromises: Promise<Result<string, string> | undefined>[] =
+      Object.entries(requestAssets).map(([key, value]) =>
+        this.handleImageUpload(value, key, requestId, organizationId)
+      );
+
+    await Promise.allSettled(uploadPromises);
+    const url = this.s3Client.getRequestResponseUrl(requestId, organizationId);
+    return await this.s3Client.store(
+      url,
+      JSON.stringify({ request: requestBody, response: responseBody })
+    );
+  }
+
+  private async handleImageUpload(
+    imageUrl: string,
+    assetId: string,
+    requestId: string,
+    organizationId: string
+  ): Promise<Result<string, string> | undefined> {
+    try {
+      let assetUploadResult: Result<string, string>;
+      if (imageUrl.startsWith("data:image/")) {
+        const [assetType, base64Data] = this.extractBase64Data(imageUrl);
+        const buffer = Buffer.from(base64Data, "base64");
+        assetUploadResult = await this.s3Client.uploadBase64ToS3(
+          buffer,
+          assetType,
+          requestId,
+          organizationId,
+          assetId
+        );
+      } else {
+        const response = await fetch(imageUrl);
+        if (!response.ok) {
+          return err(`Failed to download image: ${response.statusText}`);
+        }
+        const blob = await response.blob();
+        assetUploadResult = await this.s3Client.uploadImageToS3(
+          blob,
+          requestId,
+          organizationId,
+          assetId
+        );
+      }
+
+      if (!assetUploadResult.error) {
+        await this.saveRequestResponseAssets(assetId, requestId);
+        return { data: `<helicone-asset-id key="${assetId}"/>`, error: null };
```

Comment:
Are we doing anything with the data here?

Reply:
@colegottdank , no just passing assetId and return key.  Or did I not understand the question?